### ISSUE TITLE
Package layers - values in virtual fields are incomplete

### ISSUE THREAD

**Issue by Brent-Edwards** [qgis/QGIS#55310]
### What is the bug or the crash?

When executing the "Package layers" processing tool, layers with virtual fields are incomplete (NULL values).

### Steps to reproduce the issue

The input layers have several virtual fields with varying levels of complexity.

When exported using the Right click>Export>Save Features As method, all of the values in the virtual fields are calculated and exported correctly.

When exported using the "Package layers" processing tool, the values calculated from the virtual fields are incomplete - the first few rows are ok and the rest are NULL.

To further investigate the problem, I selected a few of the incomplete features form the source layer and re-executed the "Package layers" processing tool with the "Save only selected features" option and the output is correct -  all of the values in the virtual fields are calculated and exported correctly.

### Versions

QGIS version
3.34.0-Prizren
QGIS code revision
ffbdd67881
Qt version
5.15.3
Python version
3.9.5
GDAL/OGR version
3.7.2
PROJ version
9.3.0
EPSG Registry database version
v10.094 (2023-08-08)
GEOS version
3.12.0-CAPI-1.18.0
SQLite version
3.41.1
PDAL version
2.5.5
PostgreSQL client version
15.2
SpatiaLite version
5.1.0
QWT version
6.1.6
QScintilla2 version
2.13.4
OS version
Windows 10 Version 2009

Active Python plugins
basemesh
2.0.11
changeDataSource
3.1
DataPlotly
4.0.0
geometric_attributes
0.4
gmsh
version 0.2.2
HCMGIS
23.7.19
ImportPhotos
3.0.5
LAStools
1.4
latlontools
3.6.15
leastcostpath
1.1
openlayers_plugin
2.0.0
pcraster_tools
0.3.0
processing_fusion
3.5
QPackage
1.5
QuickMultiAttributeEdit3
version 3.0.3
quick_map_services
0.19.33
splitmultipart
1.0.0
TerrainShading
0.9.4
valuetool
3.0.17
db_manager
0.1.20
otbprovider
2.12.99
processing
2.12.99

### Supported QGIS version

- [X] I'm running a supported QGIS version according to [the roadmap](https://www.qgis.org/en/site/getinvolved/development/roadmap.html#release-schedule).

### New profile

- [ ] I tried with a new [QGIS profile](https://docs.qgis.org/latest/en/docs/user_manual/introduction/qgis_configuration.html#working-with-user-profiles)

**Comment by nyalldawson** [comment]
Do the expressions use variables?

**Comment by Brent-Edwards** [comment]
> Do the expressions use variables?

Yes, some of the expressions use variables and some also reference other virtual fields.

**Comment by nyalldawson** [comment]
Can you share a project?

**Comment by Brent-Edwards** [comment]
@nyalldawson, please let me know if you require any further information. Thanks!

**Comment by Brent-Edwards** [comment]
Any news? Is the "Feedback" label still relevant?

**Comment by Brent-Edwards** [comment]
> The QGIS project highly values your report and would love to see it addressed. However, this issue has been left in feedback mode for the last 14 days and is being automatically marked as "stale". If you would like to continue with this issue, please provide any missing information or answer any open questions. If you could resolve the issue yourself meanwhile, please leave a note for future readers with the same problem and close the issue. In case you should have any uncertainty, please leave a comment and we will be happy to help you proceed with this issue. If there is no further activity on this issue, it will be closed in a week.

Please remove the stale label as there is no missing information or unanswered questions.

**Comment by Brent-Edwards** [comment]
Please remove the stale label as there is no missing information or unanswered questions.
